addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -1357,221 +1356,295 @@ registry.background = SnippetOption.extend({
-
-        this.modal = new Dialog(null, {
-            title: _t("Background Image Sizing"),
-            $content: $(qweb.render('web_editor.dialog.background_position')),
-            buttons: [
-                {text: _t("Ok"), classes: 'btn-primary', close: true, click: _.bind(this._saveChanges, this)},
-                {text: _t("Discard"), close: true, click: _.bind(this._discardChanges, this)},
-            ],
-        }).open();
-
-        this.modal.opened().then(function () {
-            // Fetch data form $target
-            var value = ((self.$target.hasClass('o_bg_img_opt_contain')) ? 'contain' : ((self.$target.hasClass('o_bg_img_opt_custom')) ? 'custom' : 'cover'));
-            self.modal.$('> label > input[value=' + value + ']').prop('checked', true);
-
-            if (self.$target.hasClass('o_bg_img_opt_repeat')) {
-                self.modal.$('#o_bg_img_opt_contain_repeat').prop('checked', true);
-                self.modal.$('#o_bg_img_opt_custom_repeat').val('o_bg_img_opt_repeat');
-            } else if (self.$target.hasClass('o_bg_img_opt_repeat_x')) {
-                self.modal.$('#o_bg_img_opt_custom_repeat').val('o_bg_img_opt_repeat_x');
-            } else if (self.$target.hasClass('o_bg_img_opt_repeat_y')) {
-                self.modal.$('#o_bg_img_opt_custom_repeat').val('o_bg_img_opt_repeat_y');
-            }
-
-            if (self.bgPos.length > 1) {
-                self.bgPos = {
-                    x: self.bgPos[0],
-                    y: self.bgPos[1],
-                };
-                self.modal.$('#o_bg_img_opt_custom_pos_x').val(self.bgPos.x.replace('%', ''));
-                self.modal.$('#o_bg_img_opt_custom_pos_y').val(self.bgPos.y.replace('%', ''));
-            }
-            if (self.bgSize.length > 1) {
-                self.modal.$('#o_bg_img_opt_custom_size_x').val(self.bgSize[0].replace('%', ''));
-                self.modal.$('#o_bg_img_opt_custom_size_y').val(self.bgSize[1].replace('%', ''));
-            }
-
-            // Focus Point
-            self.$focus = self.modal.$('.o_focus_point');
-            self._updatePosInformation();
-
-            var imgURL = /\(['"]?([^'"]+)['"]?\)/g.exec(self.$target.css('background-image'));
-            imgURL = (imgURL && imgURL[1]) || '';
-            var $img = $('<img/>', {class: 'img img-fluid', src: imgURL});
-            $img.on('load', function () {
-                self._bindImageEvents($img);
-            });
-            $img.prependTo(self.modal.$('.o_bg_img_opt_object'));
+    backgroundType: function (previewMode, value, $opt) {
+        this.$target.toggleClass('o_bg_img_opt_repeat', value === 'repeat-pattern');
+        if (value === 'cover') {
+            // background-size: cover; is set by the css when it doesn't have the repeat class
+            this.$target.css('background-size', '');
+        } else {
+            this.$target.css('background-size', 'auto');
+        }
```

Comment:
I guess we can simply add the background-size: auto as default for the o_bg_img_opt_repeat repeat rule then. I'll do that.